### ISSUE TITLE
Response: Failed to get item.. Stop working or just me?

### ISSUE THREAD

**Issue by irk3n-dev** [flyingrub/scdl#318]
```
scdl --debug --path ~/0x0F/cdn/audio/soundcloud/ --download-archive ~/0x0F/p2p/soundcloud/history.txt -l https://soundcloud.com/user970015333/sets/girls-moaning                           
Soundcloud Downloader                                                                                                                                                                             
{'--addtimestamp': False,
 '--addtofile': False,                                                                                                                                                                            
 '--debug': True,                                                                                                                                                                                 
 '--download-archive': '/home/irk3n/0x0F/p2p/soundcloud/history.txt',                                                                                                                             
 '--error': False,                                                                                                                                                                                
 '--extract-artist': False,                                                                                                                                                                       
 '--flac': False,                                                                                                                                                                                 
 '--help': False,                                                                                                                                                                                 
 '--hide-progress': False,                                                                                                                                                                        
 '--hidewarnings': False,                                                                                                                                                                         
 '--max-size': None,                                                                                                                                                                              
 '--min-size': None,                                                                                                                                                                              
 '--no-playlist-folder': False,                                                                                                                                                                   
 '--onlymp3': False,                                                                                                                                                                              
 '--path': '/home/irk3n/0x0F/cdn/audio/soundcloud/',                                                                                                                                              
 '--remove': False,                                                                                                                                                                               
 '--version': False,                                                                                                                                                                              
 '-C': False,                                                                                                                                                                                     
 '-a': False,                                                                                                                                                                                     
 '-c': False,                                                                                                                                                                                     
 '-f': False,                                                                                                                                                                                     
 '-l': 'https://soundcloud.com/user970015333/sets/girls-moaning',                                                                                                                                 
 '-m': False,                                                                                                                                                                                     
 '-n': None,                                                                                                                                                                                      
 '-o': None,                                                                                                                                                                                      
 '-p': False,                                                                                                                                                                                     
 '-s': False,                                                                                                                                                                                     
 '-t': False,                                                                                                                                                                                     
 'me': False}                                                                                                                                                                                     
Downloading to /home/irk3n/0x0F/cdn/audio/soundcloud...
https://api.soundcloud.com/resolve?url=https://soundcloud.com/user970015333/sets/girls-moaning&client_id=a3e059563d7fd3372b49b37f00a00bcf
https://api.soundcloud.com/resolve?url=https://soundcloud.com/user970015333/sets/girls-moaning&client_id=2t9loNQH90kzJcsFCODdigxfp325aq4z
Failed to get item...
None
```

**Comment by krzemienski** [comment]
Same issue here

**Comment by viktorrko** [comment]
I'm experiencing the same issue as well. Downloading my playlist which worked just fine before.

**Comment by krzemienski** [comment]
I’m assuming it’s a sc api change or something given that it just seems to have broken, I’ll try and take a stab at debugging.

**Comment by krzemienski** [comment]
So after debugging briefly it seems that the particular api url is returning 401s, my guess is that they are no longer supporting v1 of the api? I don’t know too much about the sc api but just from reading here it looks like they moved to full o auth on v2 endpoints

**Comment by jackno** [comment]
Working on this in #319 

If you need to download something now, [check out my fork](https://github.com/jackno/scdl) until it's merged here. Authentication not tested.

**Comment by iVegas** [comment]
Manually tested, works for me. But still needs a review from the maintainer.

**Comment by viktorrko** [comment]
Great job guys! I'm glad we will (most likely) be able to keep using this software.

**Comment by krzemienski** [comment]
I mean if I was about to have to use SoundCloud to listen to music on SoundCloud while stuck in my apartment, that might have just been my tipping point. So you have no idea how much I appreciate your fork and work @jackno thank you!

**Comment by triplewy** [comment]
<code>api-v2/users/{user_id}/favorites</code> was returning 404 error for me so I had to use **track_likes** api instead

**Comment by jackno** [comment]
I'll update the PR to include the `track_likes` url. Thanks

The download URL problem was also spotted by @flyingrub here: https://github.com/flyingrub/scdl/pull/319#issuecomment-602146653
 
Soundcloud is not returning it's own v2 of the download URL. This is something that Soundcloud will have to fix but until then I'll make a hotfix branch that people can use while it's still broken.

**Comment by jackno** [comment]
The problem is not only with the api-v2 link, but that soundcloud isn't parsing any /download urls. For now, we can't download original files.

**Hotfix**: [jackno/scdl/tree/download_link_hotfix](https://github.com/jackno/scdl/tree/download_link_hotfix) This branch will ignore the original download link and just download from the stream.

**TLDR**; Use this branch if you need to download something right now.

**Comment by matixmatix** [comment]
@jackno I'm testing your hotfix on my server -- any ideas why it works in root, but not using some other user?

root:
```
root@alfonso:~# which scdl
/usr/local/bin/scdl
root@alfonso:~# scdl -l https://soundcloud.com/hypedrops/crankdat-zedd
Soundcloud Downloader
Found a track
Retrieving more info on the track
```

Other user:
```
deploy@alfonso:~$ which scdl
/usr/local/bin/scdl
deploy@alfonso:~$ scdl -l https://soundcloud.com/hypedrops/crankdat-zedd
Soundcloud Downloader
Failed to get item...
```

**Comment by jackno** [comment]
@matixmatix it looks like the original scdl is still installed. Run `pip uninstall scdl` and follow the instructions in the [README](https://github.com/jackno/scdl/blob/download_link_hotfix/README.md) of the hotfix. Let me know if that fixes it.

**Comment by matixmatix** [comment]
@jackno I already followed your instructions (thanks for providing them btw!). If I try again, it says its not installed: 

```
root@alfonso:~# pip3 uninstall scdl
Can't uninstall 'scdl'. No files were found to uninstall.
```

**Comment by jackno** [comment]
@matixmatix interesting but I can't reproduce that error. Try installing it in a venv.
```bash
cd scdl-hotfix
python3 -m venv env/
source env/bin/activate
pip install -e .
```

**Comment by matixmatix** [comment]
@jackno I think it's installed correctly. It's just that I'm getting different results using different users. Under root downloading works fine, under "deploy" user it fails to get the item.

**Comment by jackno** [comment]
@matixmatix I'm not sure at this point. You're getting that error because Soundcloud API is returning a `403 Forbidden`. Can you try installing the master branch from this repo to see if the api v1 endpoints work on your deploy user?

**Comment by matixmatix** [comment]
@jackno apologies, my install paths were all messed up! Seems to be working now.

**Comment by xiaoina** [comment]
@jackno hello my good sir. Had an issue but fixed it by doing `track = get_track_info(track['track']['id'])` at https://github.com/jackno/scdl/blob/fbe1be1ee6acf6319a290babfd107c829df7a653/scdl/scdl.py#L537

**Comment by jackno** [comment]
@xiaoina thanks, but can you describe what the issue was before you changed it?

**Comment by yclee126** [comment]
When downloading multiple tracks(Likes) with the hotfix, this error occured and none of the songs were downloaded:
```
Likes n°80 of 80
'id'
Traceback (most recent call last):
  File "c:\users\name\scdl-hotfix\scdl\scdl.py", line 358, in download
    download_track(item)
  File "c:\users\name\scdl-hotfix\scdl\scdl.py", line 537, in download_track
    track = get_track_info(track['id'])
KeyError: 'id'
```
With xiaoina's fix, I could download most of the songs. Thanks! (However, now downloading single track won't work.)
And I don't know if it's relevant to it, the "downloadable" songs were actually not downloaded with this error:
```
Likes n°75 of 80
Retrieving more info on the track
Downloading Marmalade City
Downloading the original file.
'download_url'
Traceback (most recent call last):
  File "c:\users\name\scdl-hotfix\scdl\scdl.py", line 358, in download
    download_track(item)
  File "c:\users\name\scdl-hotfix\scdl\scdl.py", line 550, in download_track
    filename = download_original_file(track, title)
  File "c:\users\name\scdl-hotfix\scdl\scdl.py", line 443, in download_original_file
    original_url = track['download_url']
KeyError: 'download_url'
```
Later I found that most of them was actually not downloadable because of the download count. (in [line 549 of scdl.py](https://github.com/jackno/scdl/blob/fbe1be1ee6acf6319a290babfd107c829df7a653/scdl/scdl.py#L549), 'has_downloads_left' check was missing) For the actual downloadable songs(like  Yerzmyey - Agartha) I tried to fix it by resolving the url with the api but it didn't work.. After giving up I forced every song to be "non-downloadable" song by editing the line 549 to `if 0:`

In summary, I came across this program a few hours ago to download all of my fav songs, and this is how I got it work.
1) "Failed to get item" error -> Installed the [hotfix](https://github.com/jackno/scdl/tree/download_link_hotfix)
2) "id" error -> xiaoina's fix (scdl.py: line 537 to `track = get_track_info(track['track']['id'])`)
3) "cannot find the path" error -> added ffmpeg's bin folder to PATH
4) Partial "download_url" error -> my own fix (scdl.py: line 549 to `if 0:`)

After all of these, I got everything downloaded-win! Cheers.

**Comment by AbdElraoufSabri** [comment]
I confirm that @yclee126 fix is working, thanks

**Comment by mXsab** [comment]
thanks a lot beautiful people, I struggled a lot to make this working as I have 0 programming skill but you were so helpful. thanks again

**Comment by guthibs** [comment]
@yclee126 fix is working indeed. Only when downloading a single track `scdl -l [url]` I get a KeyError: 'track' from line 537.

Any thoughts?

**Comment by arminbw** [comment]
@yclee126 For some reasons tracks that have `downloadable` set to TRUE are missing the `download_url` key and value pair in the json file. To ignore the downloadable boolean you may use the `--onlymp3` parameter.

**Comment by mXsab** [comment]
i have downloaded half of my 909 liked tracks and all of a sudden it stopped, any idea to resume from specified nْ# of the 909 tracks? if not i guess i will have to download all again

**Comment by jackno** [comment]
None of these fixes are passing my unit tests. `api-v2` endpoints seem to be very unstable. Maybe we should go back to using only `api`. I'll do some looking.

**Comment by jackno** [comment]
Feel free to submit a PR on the hotfix, but none of the solutions are working for me either right now. It's better to just wait until Soundcloud can stabilize their API.

**Comment by flyingrub** [comment]
https://github.com/flyingrub/scdl/commit/ebc611dbb3c1728ef0be7e3db5070067500ebf82 should fix those

**Comment by Niotferdi** [comment]
Faced the same error - "Failed to get item ..."
Links from this thread to solve the problem do not work - "page not found".
